File: examples/test_language_filter.ts
/**
 * Process user authentication and validate credentials
 * This function handles user login by checking the provided credentials
 * against the database and returns an authentication token if successful
 */
export function authenticate_user(username: string, password: string): string {
    // Implementation details
    return "token123";
}

/**
 * Fetch user profile data from the database
 * Retrieves comprehensive user information including preferences and settings
 * for the specified user identifier
 */
export function get_user_profile(userId: number): UserProfile {
    // Implementation details
    return { id: userId, name: "User" };
}

/**
 * Calculate order total with tax and shipping
 * Computes the final price including all applicable taxes and shipping costs
 * based on the user's location and selected shipping method
 */
export function calculate_order_total(items: Item[], location: string): number {
    // Implementation details
    return 42.0;
}

/**
 * Send notification email to user
 * Dispatches an email notification to the user's registered email address
 * with the specified subject and message content
 */
export function send_email_notification(email: string, subject: string, body: string): boolean {
    // Implementation details
    return true;
}

// Type definitions
interface UserProfile {
    id: number;
    name: string;
}

interface Item {
    price: number;
    quantity: number;
}

File: examples/test_language_filter.rs
/// Process user authentication and validate credentials
/// This function handles user login by checking the provided credentials
/// against the database and returns an authentication token if successful
pub fn authenticate_user(username: &str, password: &str) -> Result<String, AuthError> {
    // Implementation details
    Ok("token123".to_string())
}

/// Fetch user profile data from the database
/// Retrieves comprehensive user information including preferences and settings
/// for the specified user identifier
pub fn get_user_profile(user_id: u32) -> Result<UserProfile, DatabaseError> {
    // Implementation details
    Ok(UserProfile::default())
}

/// Calculate order total with tax and shipping
/// Computes the final price including all applicable taxes and shipping costs
/// based on the user's location and selected shipping method
pub fn calculate_order_total(items: &[Item], location: &str) -> f64 {
    // Implementation details
    42.0
}

/// Send notification email to user
/// Dispatches an email notification to the user's registered email address
/// with the specified subject and message content
pub fn send_email_notification(email: &str, subject: &str, body: &str) -> bool {
    // Implementation details
    true
}

// Dummy types for compilation
pub struct AuthError;
pub struct UserProfile;
impl Default for UserProfile {
    fn default() -> Self {
        UserProfile
    }
}
pub struct DatabaseError;
pub struct Item;
